Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIScriptingCocoa.mm

```diff
@@ -0,0 +1,264 @@
+    }
+
+    // FIXME: <https://webkit.org/b/259954> May need to remove unsupported keys.
+    // FIXME: Handle script injection in the UI Process.
+}
+
+void WebExtensionAPIScripting::insertCSS(NSDictionary *cssInfo, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
+{
+    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/insertCSS
+
+    if (!validateCSS(cssInfo, outExceptionString))
+        return;
+
+    // FIXME: <https://webkit.org/b/259954> May need to remove unsupported keys.
+    // FIXME: Handle css injection in the UI Process.
+
+}
+
+void WebExtensionAPIScripting::removeCSS(NSDictionary *cssInfo, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
+{
+    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/removeCSS
+
+    if (!validateCSS(cssInfo, outExceptionString))
+        return;
+
+    // FIXME: <https://webkit.org/b/259954> May need to remove unsupported keys.
+    // FIXME: Handle css removal in the UI Process.
+}
+
+void WebExtensionAPIScripting::registerContentScripts(NSObject *scripts, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
+{
+    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/registerContentScripts
+
+    // FIXME: <https://webkit.org/b/261769> Implement this.
+}
+
+void WebExtensionAPIScripting::getRegisteredContentScripts(NSDictionary *filter, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
+{
+    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/getRegisteredContentScripts
+
+    // FIXME: <https://webkit.org/b/261769> Implement this.
+}
+void WebExtensionAPIScripting::updateContentScripts(NSObject *scripts, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
+{
+    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/updateContentScripts
+
+    // FIXME: <https://webkit.org/b/261769> Implement this.
+}
+void WebExtensionAPIScripting::unregisterContentScripts(NSDictionary *filter, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
+{
+    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/unregisterContentScripts
+
+    // FIXME: <https://webkit.org/b/261769> Implement this.
+}
+
+bool WebExtensionAPIScripting::validateScript(NSDictionary *script, NSString **outExceptionString)
+{
+    static NSArray<NSString *> *requiredKeys = @[
+        targetKey,
+    ];
+
+    static NSDictionary<NSString *, id> *keyTypes = @{
+        argsKey: @[ NSObject.class ],
+        argumentsKey: @[ NSObject.class ],
+        filesKey: @[ NSString.class ],
+        funcKey: NSObject.class,
+        functionKey : NSObject.class,
```

Comment:
This will never fail, since everything is an `NSObject`, and `nil` values are excluded from the dictionary. These should really be `JSValue.class` at least, and `classToClassString()` needs modified to have `JSValue.class` strings. Sadly we can't get more specific with a "function" class, so the strings should be "value" or "object". A check later for `func.isObject` can at maybe least say "function".